src/debugger/nodeDebugWrapper.ts

```diff
@@ -1,194 +1,246 @@
+                                    // and the user needs to Reload JS manually. We prewarm it to prevent that issue
+                                    generator.step("prewarmBundleCache");
+                                    Log.logMessage("Prewarming bundle cache. This may take a while ...");
+                                    return nodeDebugWrapper.remoteExtension.prewarmBundleCache(nodeDebugWrapper.mobilePlatformOptions.platform);
+                                })
+                                .then(() => {
+                                    generator.step("mobilePlatform.runApp");
+                                    Log.logMessage("Building and running application.");
+                                    return mobilePlatform.runApp();
+                                })
+                                .then(() =>
+                                    nodeDebugWrapper.attachRequest(originalRequest, this, request, args, mobilePlatform));
+                        }
+                    }).catch(error =>
+                        nodeDebugWrapper.bailOut(this, error.message));
             });
+        };
+    }
 
-            // We do not permit arbitrary args to be passed to our process
-            args.args = [
-                args.platform,
-                debugServerListeningPort.toString(),
-                !isNullOrUndefined(args.iosRelativeProjectPath) ? args.iosRelativeProjectPath : IOSPlatform.DEFAULT_IOS_PROJECT_RELATIVE_PATH,
-                args.target || "simulator",
-            ];
+    /**
+     * Intecept the "attachRequest" instance method of NodeDebugSession to interpret arguments
+     */
+    private customizeAttachRequest(): void {
+        const nodeDebugWrapper = this;
+        this.nodeDebugSession.prototype.attachRequest = function (request: any, args: IAttachRequestArgs) {
+            nodeDebugWrapper.requestSetup(this, args);
+            nodeDebugWrapper.attachRequest(nodeDebugWrapper.nodeDebugSession.prototype.launchRequest, this, request, args, null);
+        };
+    }
 
-            if (!isNullOrUndefined(args.logCatArguments)) { // We add the parameter if it's defined (adapter crashes otherwise)
-                args.args = args.args.concat([parseLogCatArguments(args.logCatArguments)]);
-            }
+    /**
+     * Intecept the "disconnectRequest" instance method of NodeDebugSession to interpret arguments
+     */
+    private customizeDisconnectRequest(): void {
+        const originalRequest = this.nodeDebugSession.prototype.disconnectRequest;
+        const nodeDebugWrapper = this;
+
+        this.nodeDebugSession.prototype.disconnectRequest = function (response: any, args: any): void {
+            // First we tell the extension to stop monitoring the logcat, and then we disconnect the debugging session
+            nodeDebugWrapper.remoteExtension.stopMonitoringLogcat()

```

Comment:
Ok, then we should store the platform in the same way we store the remote extension and use it to check this.